eng/targets/Imports.targets

```diff
@@ -30,6 +30,12 @@
   <PropertyGroup Condition="'$(DisableNullableWarnings)' == 'true'">
     <NoWarn>$(NoWarn);Nullable</NoWarn>
   </PropertyGroup>
+  
+  <!-- Workaround for https://github.com/dotnet/roslyn/issues/42166 -->
+  <!-- PERF: This property group prevents the analyzers from CodeStyle NuGet package from being executed by default in CI -->
+  <PropertyGroup Condition="'$(DesignTimeBuild)' != 'true'">
```

Comment:
If we want this to be enabled for continuous integration only then we should use `'$(ContinuousIntegration)' != 'true'`. That is the standard way of conditioning data on CI.

Reply:
That would also disable all these analyzers for design time builds, we do not want that.

Reply:
Who is "we" here? This is definitely what a good portion of the compiler team wants for exactly what you laid out in this email. I want my inner loop fast including building from Visual Studio.

Reply:
design time build == live analysis during typing. If we make your suggested change, we will not execute these analyzers even during typing and see no suggestions.